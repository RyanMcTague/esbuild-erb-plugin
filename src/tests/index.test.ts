import { test, expect, describe } from '@jest/globals';
import { build } from 'esbuild';
import { erbPlugin } from '../index';
import { join } from 'path';

describe('esbuild-erb-plugin', () => {
  const bundle = async (fixture: string): Promise<string> => {
    const buildOptions = {
      entryPoints: [join(__dirname, `../fixtures/${fixture}`)],
      plugins: [erbPlugin()],
      bundle: true,
      write: false,
      outfile: 'out.js',
    };
    const result = await build(buildOptions);

    return result.outputFiles![0].text;
  };

  test('compiles code without an erb import <basic.ts>', async () => {
    const result = await bundle('basic.ts');
    expect(result).toMatchSnapshot();
  });
});
