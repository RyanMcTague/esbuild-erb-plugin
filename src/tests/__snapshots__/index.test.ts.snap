// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esbuild-erb-plugin compiles code without an erb import <basic.ts> 1`] = `
""use strict";
(() => {
  // src/fixtures/basic.ts
  function sayHello() {
    const h1 = document.createElement("h1");
    const node = document.createTextNode("Hello");
    h1.appendChild(node);
    document.appendChild(h1);
  }
  sayHello();
})();
"
`;
